python/packages/autogen-core/samples/adas/README.md

```diff
@@ -0,0 +1,515 @@
+
+### Generate new Agent Systems
+#### Prepare your dataset
+First download your dataset locally.
+
+Then create a copy the file called `utils_benchmark_template.py`, and name it with a suffix corresponding to your benchmark. For example, see `utils_drop.py`. Place this under the `adas` directory. This file will later be passed to the `benchmark_specific_utils_file` flag when running the script.
+
+Under the `load_dataset` function, add logic to load in your dataset. Do any preprocessing that is required, such as adding instructions or any few-shot examples with the actual input data.
+
+```python
+# utils_my_benchmark.py
+def load_dataset(filename: str) -> List[Dict[str, Any]]:
+  df = pd.read_csv(filename)
+  data = [{"inputs": "Your job is to solve this math problem: " + inputs, "targets": targets} for inputs, targets in df]
+  return data
+```
+#### Prepare your evaluation function
+In the same `utils_my_benchmark.py` file, add logic to the compute_metrics function to evaluate the ground truth against the predictions made by the agent system.
+
+```python
+# utils_my_benchmark.py
+def compute_metrics(predictions: List[Any], labels: List[Any]) -> List[float]:
+  return np.square(np.subtract(A, B)).mean()
+```
+### Choose the LLMs
+#### Choose the LLMs for the meta-agent
+Recommendation is GPT-4o, but you can choose whatever model you have access to on Azure.
+
+o1-preview is also reported to be great at writing code, and we suggest you try that if you have access. Caveat: Beta version has [limitations](https://platform.openai.com/docs/guides/reasoning/beta-limitations#beta-limitations) such as not supporting `SystemMessages`.
+
+This should be passed as a JSON string to the `meta_agent_model_config` flag.
+```bash
+--meta_agent_model_config='{"api_version": "2024-08-01-preview", "azure_endpoint": "https://<user>-aoai1.openai.azure.com/openai/deployments/o1-preview/chat/completions?api-version=2024-08-01-preview", "model_capabilities": {"function_calling": false, "json_output": false, "vision": false}, "azure_ad_token_provider": "DEFAULT", "model": "o1-preview-2024-09-12"}'
+```
+#### Choose the LLM for the base agents used within the agent system
+The paper authors use GPT-3.5 (for cost purposes), but we recommend GPT-4o for better quality.
+
+This should be passed as a JSON string to the `base_agent_model_config` flag.
+```bash
+--base_agent_model_config='{"api_version": "2023-03-15-preview", "azure_endpoint": "https://<user>-aoai1.openai.azure.com/openai/deployments/gpt-4o/chat/completions?api-version=2023-03-15-preview", "model_capabilities": {"function_calling": true, "json_output": true, "vision": true}, "azure_ad_token_provider": "DEFAULT", "model": "gpt-4o-2024-08-06"}'
+```
+### Run ADAS
```

Comment:
Running adas.py does seem to generate standard output and standard error output. Strangely it seems like the output in the error stream is of type info. I recommend to consolidate and also give users the option to adjust the logging threshold. I feel like we want to aim at INFO including all relevant output that allows a user to understand what ADAS is doing. Put everything else into DEBUG. Id recommend to ensure lengthy prompts are in DEBUG, where as the that the LLM was queried and gist of the prompt should be in INFO.